src/helpers/shell-history.ts

```diff
@@ -0,0 +1,52 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+
+// Function to get the history file based on the shell
+function getHistoryFile(): string | null {
+  const shell = process.env.SHELL || '';
+  const homeDir = os.homedir();
+
+  switch (path.basename(shell)) {
+    case 'bash':
+    case 'sh':
+      return path.join(homeDir, '.bash_history');
+    case 'zsh':
+      return path.join(homeDir, '.zsh_history');
+    case 'fish':
+      return path.join(homeDir, '.local', 'share', 'fish', 'fish_history');
+    case 'ksh':
+      return path.join(homeDir, '.ksh_history');
+    case 'tcsh':
+      return path.join(homeDir, '.history');
+    default:
+      return null;
+  }
+}
+
+// Function to get the last command from the history file
+function getLastCommand(historyFile: string): string | null {
+  try {
+    const data = fs.readFileSync(historyFile, 'utf8');
+    const commands = data.trim().split('\n');
+    return commands[commands.length - 1];
+  } catch (err) {
+    // Ignore any errors
+    return null;
+  }
+}
+
+// Function to append the command to the history file if it's not the same as the last command
+export function appendToShellHistory(command: string): void {
+  const historyFile = getHistoryFile();
+  if (historyFile) {
+    const lastCommand = getLastCommand(historyFile);
+    if (lastCommand !== command) {
+      fs.appendFile(historyFile, `${command}\n`, (err) => {
```

Comment:
should the newline come before the new command instead of after?
```suggestion
      fs.appendFile(historyFile, `\n${command}`, (err) => {
```

or is the idea that these files already always end in a newline (and if so are we 100% confident that this is the case all the time for all of the supported shells, or perhaps should we at least check, like

```
let fileContents = await readFile(historyFile)
if (!fileContents.endsWith('\n')) {
  // ensure we add the newline
}

```

Reply:
Thanks for the suggestion, but I added a newline at the end of the string because that's a POSIX rule. All lines in text files should end with a newline character, including the last line of plain text files. We can expect a newline at the end of all history files.

However, I am not opposed to checking it first.

Reply:
ah i see, works for me!